### PR TITLE
mirrord binary download improvements

### PR DESCRIPTION
**Pull request by infiniteregrets**
This PR:
- removes quotes around GITHUB_PATH could be a potential error
- adds a check for CI builds to choose the release builds as their tag is greater than the latest release

### CODE REVIEW

**Review by meowjesty — APPROVED** [review]
:+1:

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt [hunk 1]
```diff
@@ -78,20 +80,22 @@ class MirrordBinaryManager {
 
             val autoUpdate = MirrordSettingsState.instance.mirrordState.autoUpdate
             val userSelectedMirrordVersion = MirrordSettingsState.instance.mirrordState.mirrordVersion
+            manager.latestSupportedVersion = manager.fetchLatestSupportedVersion(product, indicator)
```
Comment (Razz4780) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt:83]: Double fetching latest supported version, second time in line 102

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt [hunk 2]
```diff
@@ -35,6 +36,7 @@ private const val DOWNLOAD_ENDPOINT = "https://github.com/metalbear-co/mirrord/r
 class MirrordBinaryManager {
     @Volatile
     private var latestSupportedVersion: String? = null
+    private var downloadVersion: String? = null
```
Comment (Razz4780) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt:39]: Is `latestSupportedVersion` used at all now?

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt [hunk 3]
```diff
@@ -278,7 +274,18 @@ class MirrordBinaryManager {
             }
 
             val binary = MirrordBinary(output)
-            if (requiredVersion == null || requiredVersion == binary.version) {
+
+            val isRequiredVersion = try {
```
Comment (Razz4780) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt:278]: This check is modified only here, what about `findBinaryInStorage`?

Comment (infiniteregrets) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt:278]: for the CI we only care about the path as we add the latest build to the path

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt (hunk 2 above)
Comment (infiniteregrets) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt:39]: both have been kept separate, as a user can specify a version like 3.60.0 and we were considering that as latestsupported version so yeah we need to keep both, but separate

**Review by Razz4780 — APPROVED** [review]

**Pull request merged**